.pre-commit-config.yaml

```diff
@@ -0,0 +1,25 @@
+# Set up locally by running
+#
+# $ pre-commit install
+#
+# See https://pre-commit.com/
+default_language_version:
+  python: python3.8
+
+repos:
+- repo: https://gitlab.com/pycqa/flake8
+  rev: 3.8.3
+  hooks:
+  - id: flake8
+- repo: https://github.com/timothycrosley/isort
+  rev: 5.5.2
+  hooks:
+    - id: isort
+- repo: https://github.com/ambv/black
+  rev: stable
+  hooks:
+    - id: black
```

Comment:
Does this only check or also reformat before committing?

Reply:
Yes, the hooks will run and `black` and `isort` and format the current commit, but won't commit those changes.

The way `pre-commit` they work is that they `git stash` everything that's not part of the commit, apply the respective tool and then check if stuff has changed. If yes, the commit is aborted. Afterwards the stash is restored to the working directory, on top of the changes made by `black` and `isort`, if possible.

The tools are not supposed to commit changes on their own; rather alert you that something's off.